src/Microsoft.Azure.WebJobs.Host/Hosting/ServiceCollectionDecorator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.WebJobs.Host.Hosting
+{
+    internal class ServiceCollectionDecorator : IServiceCollection, ITrackedServiceCollection
```

Comment:
Another tiny nit -- maybe call this `TrackedServiceCollection`? We typically don't use `Decorator`, even if the class is a decorator. Better to describe what it's being used for.

Reply:
I had that feedback, with the exact name, in a previous commend :)